metaflow/plugins/airflow/airflow_utils.py

```diff
@@ -0,0 +1,410 @@
+            else:
+                del kwargs["execution_delta"]
+    return kwargs
+
+
+def _get_sensor(name):
+    if name == SensorNames.EXTERNAL_TASK_SENSOR:
+        # ExternalTaskSensors uses an execution_date of a dag to
+        # determine the appropriate DAG.
+        # This is set to the exact date the current dag gets executed on.
+        # For example if "DagA" (Upstream DAG) got scheduled at
+        # 12 Jan 4:00 PM PDT then "DagB"(current DAG)'s task sensor will try to
+        # look for a "DagA" that got executed at 12 Jan 4:00 PM PDT **exactly**.
+        # They also support a `execution_timeout` argument to
+        from airflow.sensors.external_task_sensor import ExternalTaskSensor
+
+        return ExternalTaskSensor
+    elif name == SensorNames.S3_SENSOR:
+        try:
+            from airflow.providers.amazon.aws.sensors.s3 import S3KeySensor
+        except ImportError:
+            raise AirflowSensorNotFound(
+                "This DAG requires a `S3KeySensor`. "
+                "Install the Airflow AWS provider using : "
+                "`pip install apache-airflow-providers-amazon`"
+            )
+        return S3KeySensor
+    elif name == SensorNames.SQL_SENSOR:
+        from airflow.sensors.sql import SqlSensor
+
+        return SqlSensor
+
+
+class AirflowTask(object):
+    def __init__(self, name, operator_type="kubernetes", flow_name=None):
+        self.name = name
+        self._operator_args = None
+        self._operator_type = operator_type
+        self._flow_name = flow_name
+
+    def set_operator_args(self, **kwargs):
+        self._operator_args = kwargs
+        return self
+
+    def _make_sensor(self):
+        TaskSensor = _get_sensor(self._operator_type)
+        return TaskSensor(
+            task_id=self.name,
+            **_parse_sensor_args(self._operator_type, self._operator_args)
+        )
+
+    def to_dict(self):
+        return {
+            "name": self.name,
+            "operator_type": self._operator_type,
+            "operator_args": self._operator_args,
+        }
+
+    @classmethod
+    def from_dict(cls, jsd, flow_name=None):
+        op_args = {} if not "operator_args" in jsd else jsd["operator_args"]
+        return cls(
+            jsd["name"],
+            operator_type=jsd["operator_type"]
+            if "operator_type" in jsd
+            else "kubernetes",
+            flow_name=flow_name,
+        ).set_operator_args(**op_args)
+
+    def _kubenetes_task(self):
+        try:
+            from airflow.contrib.operators.kubernetes_pod_operator import (
+                KubernetesPodOperator,
+            )
+        except ImportError:
+            try:
+                from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import (
+                    KubernetesPodOperator,
+                )
+            except ImportError as e:
+                raise KubernetesProviderNotFound(
+                    "This DAG requires a `KubernetesPodOperator`. "
```

Comment:
`This DAG utilizes KubernetesPodOperator. Please install the Airflow Kubernetes provider using '%s -m pip install apache-airflow-providers-cncf-kubernetes'` % sys.executable